### PR TITLE
Apply border-radius to img, canvas, iframe

### LINKED ISSUE #17510: Servo doesn't properly render img border-radius
Example on: https://github.com/business

The face pictures are supposed to be circles but in Servo they are squares.

Image (Servo on top, Firefox on bottom):

border-radius

The offending line appears to be: ``

Tested in Windows 10.

### PR DESCRIPTION
**Pull request by streichgeorg**
Added some code that adjusts the clipping region of ```img```, ```canvas```, ```iframe``` elements for the ```border-radius``` property. 

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #17510 (github issue number if applicable).

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17589)

### CODE REVIEW

File: components/layout/display_list_builder.rs [hunk 1]
```diff
@@ -830,18 +830,25 @@ fn convert_ellipse_size_keyword(keyword: ShapeExtent,
     }
 }
 
+fn clip_for_border_radius(style: &ServoComputedValues,
+                          clip: &Rect<Au>) -> ClippingRegion {
+    let border_radii = build_border_radius(clip, style.get_border());
+    let mut adjusted_clip = ClippingRegion::max();
+    if !border_radii.is_square() {
+        adjusted_clip.intersect_with_rounded_rect(clip, &border_radii);
+    };
```
Comment (emilio) [inline components/layout/display_list_builder.rs]: nit: This semicolon is not needed, please remove it.

File: components/layout/display_list_builder.rs [hunk 2]
```diff
@@ -1834,6 +1841,8 @@ impl FragmentDisplayListBuilding for Fragment {
         let stacking_relative_content_box =
             self.stacking_relative_content_box(stacking_relative_border_box);
 
+        let clipping_region = clip_for_border_radius(&self.style, &stacking_relative_content_box);
```
Comment (emilio) [inline components/layout/display_list_builder.rs]: Why `stacking_relative_border_box`? I'd have expected this to use `clip`.

Also, what about the other uses for the `clip` rect? Should they account for border-radius too?

**Comment by emilio** [comment]
Probably @glennw can review this faster than I, too :)

File: components/layout/display_list_builder.rs (hunk 2 above)
Comment (streichgeorg) [inline components/layout/display_list_builder.rs]: I tried ```clip``` and it didn't work, but strangely it works with ``stacking_relative_content_box``` so I just used that. I don't really have any oversight over this code so I don't know why.

**Comment by streichgeorg** [comment]
I won't be able to do any work next week, so if there are any changes required I'll do them next weekend.

**Comment by jdm** [comment]
You will need to run `./mach update-manifest` to avoid the error that was reported.

**Comment by glennw** [comment]
This looks good to me, but I'd like @mrobinson to check over it too.

**Comment by mrobinson** [comment]
Oof! I was working on this one in parallel (https://github.com/mrobinson/servo/tree/rounded-corners). The reason it isn't working for iframes is that I fixed this in WebRender and I was waiting for the WebRender update to post my PR. There's also an issue here with the size of the clip. We really want to use the inner region. I'm really sorry for not noticing this PR sooner.

**Comment by streichgeorg** [comment]
Oh, I probably should have looked around a bit more before working on this. I guess this issue is a lot more complex than I thought it was, so I'll just close this PR.

**Pull request closed without merging**